rfcs/rfc-0000-discriminators.md

```diff
@@ -0,0 +1,208 @@
+             ^ separator between chosen name and discriminator
+              ^ discriminator
+```
+
+Discriminators for new users are generated randomly on sign up while existing users will
+get a randomly generated discriminators upon the server software being upgraded.
+
+> Revolt treats usernames as case-insensitive so a full username such as "Jason#1234" is
+> effectively the same as "jasON#1234" but only "Jason#1234" will display in the UI.
+
+Users may:
+
+- Change the case of their username at any moment, such as from "Jason" to "jasON".
+
+  This will never affect their discriminator.
+
+- Change their username, but may have their discriminator regenerated if there is a conflict.
+
+  Such as if you are changing from "Jason#1234" to "Phil", but "Phil#1234" is taken.
+
+### User Opinion
+
+Prior discussions on the forums primarily seem to lean in favour of discriminators being
+implemented into Revolt, the internal team has not raised any complaints against it, and
+the primary community on Revolt ("Revolt Lounge") appears to be somewhat split on the
+issue.
+
+As a recent case study, Discord's change to unique usernames only has generated a lot of
+backlash and criticism from the community, it appears to be overwhemingly negative but
+these experiences are mostly coming from long-term users of Discord. The true opinion of
+the average user has not been gauged yet.
+
+### Usability
+
+This may make it more difficult to understand how to add users on Revolt, however this
+shouldn't be an issue as long as it's clearly communicated through the user interface.
+Revolt already allows a wide range of Unicode in usernames so we are definitely not
+making a significant UX hit towards the usability of the friend system.
+
+> There are further considerations listed at the end of this document which must be
+> addressed if this RFC were to be implemented.
+
+# Reference-level explanation
+
+WIP
+
+This is the technical section of the RFC, it should go over in detail:
+
+- Its interaction with other features
+- How this will be implemented
+- Corner or edge cases
+
+This section should reference the examples in the previous section and disect them in more detail.
+
+# Drawbacks
+
+The main drawbacks are:
+
+- Additional information to remember about your username
+
+  Although this works both ways in the sense that, you can have a much simpler username.
+
+- Users may no longer have a completely 'unique' username on the platform
+
+  But we're approaching this from the standpoint that usernames on platforms such as these should not have to compete for 'uniqueness'.
+
+I want to take this section to also discuss the case study of Discord's removal of the discriminator system and their [cited drawbacks](https://discord.com/blog/usernames#heading-3):
+
+- > More than 40% of you either don’t remember your discriminator or don’t even know what a discriminator is. That’s a big problem when discriminators are required to add a new friend.
```

Comment:
^ this. though i would argue "people not remembering their discriminators" is *definitely* a thing that happens. 

that said, either of these issues could easily be mitigated with some relatively simple UI/UX work. hell, even discord themselves already have a partial solution to this by letting you copy your entire username + discriminator with a single click... but even that might not be a well-known feature because you have to click on your own username at the bottom of the left side panel. i guarantee most people have no idea you can even do this because it's not mentioned or explained anywhere.

Reply:
I tried to be clear that I'm quoting them here, but I did want to argue against their points at face value.

Reply:
Added a note regarding this.